x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/SequenceKey.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.eql.execution.sequence;
+
+import org.elasticsearch.common.util.CollectionUtils;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+public class SequenceKey {
+
+    public static final SequenceKey NONE = new SequenceKey();
+
+    private final Object[] keys;
+
+    public SequenceKey(Object... keys) {
+        this.keys = keys;
+    }
+
+    public List<String> asStringList() {
+        String[] s = new String[keys.length];
```

Comment:
Because this achieves the same thing but faster. By using `asList` the backing array is returned with a fixed size (no extra allocation) and wrapped as a semi-immutable list, meaning it won't grow but you can replace it.
Which I think it's fine considering the internal usage.